sympy/sets/sets.py

```diff
@@ -738,6 +742,9 @@ def __new__(cls, *sets, **assumptions):
         if len(sets) == 0:
             return FiniteSet(())
 
+        if len(sets) == 1 and isinstance(sets[0], cls):
+            return sets[0]
```

Comment:
I don't think this is right...

Reply:
This change to ProductSet breaks the key invariant in the current definition of ProductSet. I should have documented the definition but didn't but here are key invariants as implemented on master:
```
ProductSet(A, B, C) is the set of all 3-tuples (a,b,c) where a in A, b in B and c in C.
ProductSet(*(S**n)) is the set of n-tuples of elements of S
```
In code that second proposition is:
```
In [9]: for n in range(10): 
   ...:     print(n, (0,)*n in Reals**n) 
   ...:                                                                                                                                                       
0 True
1 True
2 True
3 True
4 True
5 True
6 True
7 True
8 True
9 True
```
The change here breaks that in the case of tuples of tuples:
```julia
In [10]: 0 in Reals                                                                                                                                           
Out[10]: True

In [11]: (0,) in Reals**1                                                                                                                                     
Out[11]: True

In [12]: ((0,),) in (Reals**1)**1                                                                                                                             
Out[12]: False
```
That last result gives True on master